lib/iris/_concatenate.py

```diff
@@ -926,6 +930,12 @@ def register(
                         or candidate_axis not in coord_b.dims
                     ):
                         if not coord_a == coord_b:
+                            mismatch_error_msg = (
+                                "Auxiliary coordinates differ for phenomenon"
+                                f" {self._cube_signature.defn.name()}:\n"
```

Comment:
```suggestion
                                f" `{self._cube.name()}`:\n"
```

We were doing our best to mimic `match()`, but we have access to extra info here that makes for more readable code.

I feel the backticks help highlight the name within the message, too.

Obvs this comment applies to the other cases too.